### PR TITLE
T 11017563 create minimized card

### PR DESCRIPTION
**Pull request by evnikitin**
The following changes have been made in this PR:
- added a minimized card instead of a common card;
- replaced the power indicator to allow for the option of adding images instead of text. This was necessary due to the use of special cards;
-  changed the styles of certain indicators and containers for the purpose of reusability.

### CODE REVIEW

File: src/components/MainLayout/MainPage/CommonGameArea/UserCards/UserCards.tsx [hunk 1]
```diff
@@ -1,6 +1,7 @@
 import React from 'react';
 import { UserCardsWrapperStyled } from './UserCardsWrapperStyled';
-import { Card } from '../../../../Shared/Card';
+// import { Card } from '../../../../Shared/Card'; если нужно, то можно поставить карту с описанием
```
Comment (solovevayaroslavna) [inline src/components/MainLayout/MainPage/CommonGameArea/UserCards/UserCards.tsx]: do we need this comment?

File: src/components/Shared/Card/CardComponentsStyled.ts [hunk 2]
```diff
@@ -78,13 +82,23 @@ export const CardBody = styled.div <{ backgroundImage: string }>`
...
+`;
+
 export const TypeOfArmyStyled = styled.div`
   position: absolute;
   top: 10%;
   left: 10%;
   width: 80%;
   background-color: red;
   height: 90%;
+  z-index:-1;
```
Comment (solovevayaroslavna) [inline src/components/Shared/Card/CardComponentsStyled.ts]: why we need -1?

File: src/utils/Images.utils.ts [hunk 3]
```diff
@@ -58,7 +62,7 @@ function createCards() {
       id: 5,
       amount: 2,
       ability: 2,
-      power: 2,
+      power: one,
```
Comment (solovevayaroslavna) [inline src/utils/Images.utils.ts]: why power is string?
I don't think that BE will send to us something else besides numbers

File: src/utils/Images.utils.ts [hunk 4]
```diff
@@ -78,41 +82,41 @@ function createCards() {
...
+      id: 10,
       amount: 4,
-      ability: 1,
-      power: 11,
-      category: 1,
-      header: 'Гребанная пехтура капитана',
-      text: 'Пожертвуйте грошик ветерану Бренны',
-      image: `${hero1}`,
+      ability: null,
+      power: snowflake,
```
Comment (solovevayaroslavna) [inline src/utils/Images.utils.ts:102]: the weather card is another type of card, so....
as example

type CardType: {
id: number/string,
power: number,
header: string,
text: string
}
type WeatherCard: {
id: ...
header:...
text:...
}

const arrayOfAllCards: Array<CardType|WeatherCard> = [ usualCard, weatherCard]

Comment (solovevayaroslavna) [reply src/utils/Images.utils.ts:102]: or you can use power?: number as unrequired field, but better to use different interfaces like

interface CommonCard = {
props, which we have in all type of cards
}

then 
interface ArmyCard extends CommonCard {
props for ArmyCard, which we haven't in common card
}

then interface WeatherCard extends CommonCard {
props for weather card, which we haven't in common card
}

**Review by solovevayaroslavna — CHANGES_REQUESTED** [review]

File: src/components/MainLayout/MainPage/CommonGameArea/UserCards/UserCards.tsx (hunk 1 above)
Comment (evnikitin) [reply src/components/MainLayout/MainPage/CommonGameArea/UserCards/UserCards.tsx]: not needed, will be removed

File: src/components/Shared/Card/CardComponentsStyled.ts (hunk 2 above)
Comment (evnikitin) [reply src/components/Shared/Card/CardComponentsStyled.ts]: I use -1 to not set z-index values for all components, but only for the TypeOfArmyStyled component, which is always positioned behind other components.

Comment (solovevayaroslavna) [reply src/components/Shared/Card/CardComponentsStyled.ts]: https://techrocks.ru/2021/09/18/how-to-manage-z-index-in-css/ please read this article and create a function with jeneration of z indexes. just number -1 is not a good practice

File: src/utils/Images.utils.ts (hunk 4 above)
Comment (solovevayaroslavna) [reply src/utils/Images.utils.ts:102]: it is still incorrect
to incorrectly specify power in both types as the same field (or rather, it is technically possible and in some cases applicable, but in our case it is logically incorrect)

it is more correct to remove power as a property from the weather card and add type to weater card

as you can see i removed power in previous comment from weatherCard
image
so if we need to detect wich weather card we have it's usful to use type: snow, where snow is also enum of cards

Comment (solovevayaroslavna) [reply src/utils/Images.utils.ts:102]: but, to be honest it depends more from api, so i would suggest to leave it as it is and wait for api

Comment (solovevayaroslavna) [reply src/utils/Images.utils.ts:102]: so this thread we can leave as resolved

File: src/components/Shared/Card/CardComponentsStyled.ts (hunk 2 above)
Comment (evnikitin) [reply src/components/Shared/Card/CardComponentsStyled.ts]: function has been added

**Review by solovevayaroslavna — APPROVED** [review]

**Pull request merged**